datasets/motordatavoyager4.py

```diff
@@ -0,0 +1,499 @@
+                                       MotorDataVoyager4.faulty_files_folder)
+        if not os.path.isdir(healthy_data_dir):
+            raise FileNotFoundError(
+                f'\nHealthy dataset directory {healthy_data_dir} does not exist.\n')
+
+        if not os.path.isdir(faulty_data_dir):
+            raise FileNotFoundError(
+                f'\nFaulty dataset directory {faulty_data_dir} does not exist.\n')
+
+        with os.scandir(healthy_data_dir) as it:
+            if not any(it):
+                raise FileNotFoundError(
+                    f'\nHealthy dataset directory {healthy_data_dir} is empty.\n')
+
+        with os.scandir(faulty_data_dir) as it:
+            if not any(it):
+                raise FileNotFoundError(
+                    f'\nFaulty dataset directory {healthy_data_dir} is empty.\n')
+
+        # Generate Normal Features:
+        train_features = []
+        train_speeds = []
+        train_loads = []
+
+        test_normal_features = []
+        test_normal_speeds = []
+        test_normal_loads = []
+
+        for file in sorted(os.listdir(healthy_data_dir)):
+            full_path = os.path.join(healthy_data_dir, file)
+
+            # Users can parse raw data information and keep vibration speed and load properties
+            file_speed = 0
+            file_load = 0
+
+            healthy_raw_data = \
+                MotorDataVoyager4.parse_ADXL382_and_return_raw_data(full_path)
+
+            healthy_raw_data_windows = \
+                cbm_dataframe_parser.CbM_DataFrame_Parser.split_file_raw_data(
+                    healthy_raw_data,
+                    MotorDataVoyager4.sampling_rate_in_Hz_ADXL382,
+                    MotorDataVoyager4.raw_data_window_duration_in_sec,
+                    MotorDataVoyager4.overlap_ratio)
+
+            file_cnn_signals = \
+                MotorDataVoyager4.process_raw_data_windows(healthy_raw_data_windows)
+
+            num_training = int(self.train_ratio * file_cnn_signals.shape[0])
+
+            for i in range(file_cnn_signals.shape[0]):
+                if i < num_training:
+                    train_features.append(file_cnn_signals[i])
+                    train_speeds.append(file_speed)
+                    train_loads.append(file_load)
+                else:
+                    test_normal_features.append(file_cnn_signals[i])
+                    test_normal_speeds.append(file_speed)
+                    test_normal_loads.append(file_load)
+
+        train_features = np.asarray(train_features)
+        test_normal_features = np.asarray(test_normal_features)
+
+        # Generate Anomaly Features:
+        anomaly_features = []
+        test_anormal_speeds = []
+        test_anormal_loads = []
+
+        for file in sorted(os.listdir(faulty_data_dir)):
+            full_path = os.path.join(faulty_data_dir, file)
+
+            file_speed = file.split("_")[8][:2]  # Fan level like: L0, L1, H0, H1 etc
```

Comment:
We should not have any dependency on the file naming. Please remove any speed and load naming dependency.